coderd/telemetry/telemetry_test.go

```diff
@@ -110,21 +110,32 @@ func TestTelemetry(t *testing.T) {
 		_ = dbgen.User(t, db, database.User{
 			Email: "kyle@coder.com",
 		})
-		_, snapshot := collectSnapshot(t, db)
+		_, snapshot := collectSnapshot(t, db, nil)
 		require.Len(t, snapshot.Users, 1)
 		require.Equal(t, snapshot.Users[0].EmailHashed, "bb44bf07cf9a2db0554bba63a03d822c927deae77df101874496df5a6a3e896d@coder.com")
 	})
+	t.Run("Experiments", func(t *testing.T) {
+		t.Parallel()
+
+		const expName = "my-experiment"
+		exps := []string{expName}
+		_, snapshot := collectSnapshot(t, dbmem.New(), func(opts telemetry.Options) telemetry.Options {
+			opts.Experiments = exps
+			return opts
+		})
+		require.Equal(t, []telemetry.Experiment{{Name: expName}}, snapshot.Experiments)
+	})
 }
 
 // nolint:paralleltest
 func TestTelemetryInstallSource(t *testing.T) {
 	t.Setenv("CODER_TELEMETRY_INSTALL_SOURCE", "aws_marketplace")
 	db := dbmem.New()
-	deployment, _ := collectSnapshot(t, db)
+	deployment, _ := collectSnapshot(t, db, nil)
 	require.Equal(t, "aws_marketplace", deployment.InstallSource)
 }
 
-func collectSnapshot(t *testing.T, db database.Store) (*telemetry.Deployment, *telemetry.Snapshot) {
+func collectSnapshot(t *testing.T, db database.Store, addOptionsFn func(opts telemetry.Options) telemetry.Options) (*telemetry.Deployment, *telemetry.Snapshot) {
```

Comment:
suggestion, nonblocking: if you make `addOptionsFn` variadic you can avoid the need for changes on line 88 and 113

Reply:
It's an interesting idea, but I think it'd change the semantics a bit.